taichi/codegen/codegen_opengl.cpp

```diff
@@ -812,33 +812,33 @@ int _rand_i32()\n\
     if (loop_var->ret_type.data_type == DataType::i32) {
       if (!for_stmt->reversed) {
         emit("for (int {}_ = {}; {}_ < {}; {}_ = {}_ + {}) {{",
-             loop_var->raw_name(), for_stmt->begin->raw_name(),
-             loop_var->raw_name(), for_stmt->end->raw_name(),
-             loop_var->raw_name(), loop_var->raw_name(), 1);
-        // variable named `loop_var->raw_name()` is already allocated by alloca
-        emit("  {} = {}_;", loop_var->raw_name(), loop_var->raw_name());
+             loop_var->short_name(), for_stmt->begin->short_name(),
+             loop_var->short_name(), for_stmt->end->short_name(),
+             loop_var->short_name(), loop_var->short_name(), 1);
+        // variable named `loop_var->short_name()` is already allocated by alloca
+        emit("  {} = {}_;", loop_var->short_name(), loop_var->short_name());
       } else {
         // reversed for loop
         emit("for (int {}_ = {} - 1; {}_ >= {}; {}_ = {}_ - {}) {{",
-             loop_var->raw_name(), for_stmt->end->raw_name(),
-             loop_var->raw_name(), for_stmt->begin->raw_name(),
-             loop_var->raw_name(), loop_var->raw_name(), 1);
-        emit("  {} = {}_;", loop_var->raw_name(), loop_var->raw_name());
+             loop_var->short_name(), for_stmt->end->short_name(),
+             loop_var->short_name(), for_stmt->begin->short_name(),
+             loop_var->short_name(), loop_var->short_name(), 1);
+        emit("  {} = {}_;", loop_var->short_name(), loop_var->short_name());
       }
     } else {
       TI_ASSERT(!for_stmt->reversed);
       const auto type_name = opengl_data_type_name(loop_var->element_type());
       emit("for ({} {} = {}; {} < {}; {} = {} + 1) {{", type_name,
-           loop_var->raw_name(), for_stmt->begin->raw_name(),
-           loop_var->raw_name(), for_stmt->end->raw_name(),
-           loop_var->raw_name(), loop_var->raw_name());
+           loop_var->short_name(), for_stmt->begin->short_name(),
+           loop_var->short_name(), for_stmt->end->short_name(),
+           loop_var->short_name(), loop_var->short_name());
     }
     for_stmt->body->accept(this);
     emit("}}");
   }
 
   void visit(WhileControlStmt *stmt) override {
-    emit("if ({} == 0) break;", stmt->cond->raw_name());
+    emit("if ({} == 0) break;", stmt->cond->short_name());
```

Comment:
Here, instead of `tmp1/tmp2`, use `A/B` for variable name instead (like base64). NV GLSL compiler seems really cares variable length... when I use `tmp1/tmp2`, it just stucked at compiling stage and 100% cpu usage...
Not sure if CUDA/Metal backend want this, inputs are welcome @k-ye @yuanming-hu!

Reply:
Interesting - I don't have a clear clue why the NV GLSL compiler is so sensitive to variable lengths. CUDA uses LLVM so I don't worry too much about it.